### PR TITLE
Use ssl verification in tests

### LINKED ISSUE #927: SSL tests verification
Our ssl tests currently dont verify anything :)

Note for the future: maybe Requests didn't support it, but now with HTTPX I think we could pass `verify=<ca_cert>`, rather than disabling SSL verification.

_Originally posted by @florimondmanca in https://github.com/encode/uvicorn/pull/917#discussion_r549130563_

### PR DESCRIPTION
**Pull request by euri10**
Fixes #927 

begin to love httpx more and more, so easy to work with !

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -50,3 +52,10 @@ def tls_ca_certificate_private_key_encrypted_path(tls_certificate_authority):
 def tls_certificate_pem_path(tls_certificate):
     with tls_certificate.private_key_and_cert_chain_pem.tempfile() as cert_pem:
         yield cert_pem
+
+
+@pytest.fixture
+def ssl_ctx(tls_certificate):
```
Comment (florimondmanca) [inline tests/conftest.py]: (Nit) What do you think about naming it `tls_ca_ssl_context`, for consistency with other TLS fixtures that start with `tls_ca_`?

File: tests/conftest.py [hunk 2]
```diff
@@ -50,3 +52,10 @@ def tls_ca_certificate_private_key_encrypted_path(tls_certificate_authority):
...
+    ssl_ctx = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
```
Comment (florimondmanca) [inline tests/conftest.py]: I don't think there's actually a need for specifying anything, is there?

```suggestion
    ssl_ctx = ssl.SSLContext()
```

**Review by florimondmanca — COMMENTED** [review]
Sweet!

File: tests/conftest.py (hunk 1 above)
Comment (euri10) [reply tests/conftest.py]: Indeed ! done

File: tests/conftest.py (hunk 2 above)
Comment (euri10) [reply tests/conftest.py]: it seems like it doenst

**Review by florimondmanca — APPROVED** [review]
💯

**Pull request merged**